packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -393,4 +429,50 @@ export class SnapInterfaceController extends BaseController<
       (id: string) => this.messagingSystem.call('SnapController:get', id),
     );
   }
+
+  #_onNotificationsListUpdated(notificationsList: Record<string, any>[]) {
+    const snapNotificationsWithInterface = notificationsList.filter(
+      (notification) => {
+        return notification.type === 'snap' && notification.data?.detailedView;
+      },
+    );
+
+    const interfaceIdSet = new Set(
+      snapNotificationsWithInterface.map(
+        (notification) => notification.data.detailedView.interfaceId,
+      ),
+    );
+
+    const updatedState = Object.entries(this.state.interfaces).reduce<
+      Record<string, StoredInterface>
+    >((newState, [id, snapInterface]) => {
+      if (snapInterface.contentType === ContentType.Notification) {
+        if (interfaceIdSet.has(id)) {
+          newState[id] = snapInterface;
+        }
+      } else {
+        newState[id] = snapInterface;
+      }
+      return newState;
+    }, {});
+
+    this.update((state) => {
+      // @ts-expect-error - TS2589: Type instantiation is excessively deep and
+      // possibly infinite.
+      state.interfaces = updatedState;
+    });
+  }
```

Comment:
Can we update this to simply delete the unnecessary interfaces instead of recreating the entire object? With the current approach this would be a full replace which will be more expensive on the rendering side.